onyo/lib/onyo.py

```diff
@@ -85,12 +85,19 @@ def templates(self) -> set[Path]:
 
         return self._templates
 
-    def clear_caches(self) -> None:
-        self._assets = None
-        self._dirs = None
-        self._files = None
-        self._gitfiles = None
-        self._templates = None
+    def clear_caches(self, assets: bool = True, dirs: bool = True,
+                     files: bool = True, gitfiles: bool = True,
+                     templates: bool = True) -> None:
+        if assets and self._assets is not None:
```

Comment:
IMO, there's no need to check if `is not None`. It doesn't protect anything nor saves computational effort.

Reply:
> Writting None to a variable that is already None costs runtime, I think.

I would be surprised if that is the case with `None`. I don't intend to bike-shed this, but IMO, the current code is more verbose, and not-protecting against a significant runtime cost.

If you profile that chunk of code, I would be interested in seeing the results.

Reply:
After profiling, I give up the argument, since it does not change run-time. I will remove the unneeded check for `None` like proposed by you 👍

Reply:
Thanks for checking! :-D 

It would have been cool to see the results. In the future, please share!